macros/core/src/program/args.rs

```diff
@@ -0,0 +1,115 @@
+use std::collections::BTreeSet;
+use syn::{
+    parse::{Parse, ParseStream},
+    punctuated::Punctuated,
+    Ident, Path, Token,
+};
+
+#[derive(Debug, Default, PartialEq)]
+pub(super) struct ProgramArgs {
+    handle_reply: Option<Path>,
+    handle_signal: Option<Path>,
+}
+
+impl ProgramArgs {
+    pub fn handle_reply(&self) -> &Option<Path> {
+        &self.handle_reply
+    }
+
+    pub fn handle_signal(&self) -> &Option<Path> {
+        &self.handle_signal
+    }
+}
+
+impl Parse for ProgramArgs {
+    fn parse(input: ParseStream<'_>) -> syn::Result<Self> {
+        let punctuated: Punctuated<ProgramArg, Token![,]> = Punctuated::parse_terminated(input)?;
+        let mut attrs = ProgramArgs {
+            handle_reply: None,
+            handle_signal: None,
+        };
+        let mut existing_attrs = BTreeSet::new();
+
+        for ProgramArg { name, path, .. } in punctuated {
+            let name = name.to_string();
+            if existing_attrs.contains(&name) {
+                return Err(syn::Error::new_spanned(name, "parameter already defined"));
+            }
+
+            match &*name {
+                "handle_reply" => {
+                    attrs.handle_reply = Some(path);
+                }
+                "handle_signal" => {
+                    attrs.handle_signal = Some(path);
+                }
+                _ => return Err(syn::Error::new_spanned(name, "unknown parameter")),
```

Comment:
I would abort here as this is the place where we know the proper error context

Reply:
Spanned error is returned here